okio/src/appleMain/kotlin/okio/-ApplePosixVariant.kt

```diff
@@ -15,27 +15,48 @@
  */
 package okio
 
+import kotlinx.cinterop.ByteVar
 import kotlinx.cinterop.alloc
+import kotlinx.cinterop.allocArray
+import kotlinx.cinterop.get
 import kotlinx.cinterop.memScoped
 import kotlinx.cinterop.ptr
+import kotlinx.cinterop.toKString
+import okio.Path.Companion.toPath
+import okio.internal.toPath
 import platform.posix.ENOENT
 import platform.posix.S_IFDIR
+import platform.posix.S_IFLNK
 import platform.posix.S_IFMT
 import platform.posix.S_IFREG
 import platform.posix.errno
+import platform.posix.lstat
+import platform.posix.readlink
 import platform.posix.stat
 
 @ExperimentalFileSystem
 internal actual fun PosixFileSystem.variantMetadataOrNull(path: Path): FileMetadata? {
   return memScoped {
     val stat = alloc<stat>()
-    if (platform.posix.lstat(path.toString(), stat.ptr) != 0) {
+    if (lstat(path.toString(), stat.ptr) != 0) {
       if (errno == ENOENT) return null
       throw errnoToIOException(errno)
     }
+
+    var symlinkTarget: Path? = null
+    if (stat.st_mode.toInt() and S_IFMT == S_IFLNK) {
+      // `path` is a symlink, let's resolve its target.
+      memScoped {
+        val buffer = allocArray<ByteVar>(4096)
+        readlink(path.toString(), buffer, 4096)
+        symlinkTarget = buffer.toKString().toPath()
```

Comment:
4096 is somewhat arbitrary

Reply:
Also, if the return value is -1, you should do this:
```
      throw errnoToIOException(errno)
```
https://pubs.opengroup.org/onlinepubs/009696699/functions/readlink.html

Reply:
Also I think it’s nicer to pass the result to toKString():

```
symlinkTarget = buffer.toKString(endIndex = byteCount).toPath()
```

Reply:
Replace 4096 with `platform.posix.PATH_MAX`